skimage/transform/steerable.py

```diff
@@ -0,0 +1,339 @@
+from __future__ import division
+import numpy as np
+from .._shared.utils import assert_nD
+from ..util import img_as_float
+
+
+def build_steerable(image, height=5):
+    """
+    Parameters
+    ----------
+    image : 2-D array
+        input image
+    height : integer, optional
+        Height of the steerable decomposition. This includes the counting of
+        low pass and high pass subbands.
+
+    Returns
+    -------
+    coeff : subbands of Steerable decomposition,
+            stored as a list of 'height' sublists,
+            Sublists correspond to decreasing radius level in Steerable pyramid
+            The first sublist contains the high pass subband.
+            The last sublist contains the low pass subband.
+            Intermediate sublists contains subbands from different orientations
+
+    References
+    ----------
+    """
+    assert_nD(image, 2)
+
+    s = Steerable(height)
+    return s.build_scf_pyramid(image)
+
+
+def recon_steerable(coeff):
+    """
+    Parameters
+    ----------
+    coeff : subbands of Steerable decomposition,
+            stored as a list of 'height' sublists,
+            Sublists correspond to decreasing radius level in Steerable pyramid
+            The first sublist contains the high pass subband.
+            The last sublist contains the low pass subband.
+            Intermediate sublists contains subbands from different orientations
+    Returns
+    -------
+    image : 2-D array
+        reconstructed image from subbands of Steerable Pyramid decomposition
+
+    References
+    ----------
+    """
+    height = len(coeff)
+    s = Steerable(height)
+    return s.recon_scf_pyramid(coeff)
+
+
+class Steerable:
+    """Steerable Pyramid: a translation and rotation invariant free wavelet
+    """
+
+    def __init__(self, height=5):
+        """
+        Parameters
+        ----------
+        height : height of the Steerable Decomposition
+        (including high pass and low pass)
+        """
+        self.nbands = 4
+        self.height = height
+        self.isSample = True
+
+    def build_scf_pyramid(self, im):
+        """
+        Parameters
+        ----------
+        im : 2-D array
+        """
+        assert_nD(im, 2)
+
+        im = img_as_float(im)
+
+        M, N = im.shape
+        log_rad, angle = base(M, N)
+        Xrcos, Yrcos = rcos_curve(1, -0.5)
+        Yrcos = np.sqrt(Yrcos)
+        YIrcos = np.sqrt(1 - Yrcos * Yrcos)
+
+        lo0mask = point_op(log_rad, YIrcos, Xrcos)
+        hi0mask = point_op(log_rad, Yrcos, Xrcos)
+
+        imdft = np.fft.fftshift(np.fft.fft2(im))
```

Comment:
Perhaps all `fft2` and `ifft2` calls should use the `norm` argument via a class parameter such as `self.fft_norm`.  By default the numpy FFTs use `norm=None`, but `norm='oath'` might be desirable to have energy in the coefficients that is similar to that of the input image.